### PR TITLE
Feature/elm integration

### PR DESCRIPTION
**Pull request by tdroxler**
I'll do a demo at the office :-)

### CODE REVIEW

File: sarsi-elm/Main.hs [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import System.Exit (ExitCode, exitWith)
+
+import qualified Data.List as List
+import qualified Data.Text as Text
+import qualified Sarsi as Sarsi
+
+title :: String
+title = concat [Sarsi.title, "-elm"]
+
+-- TODO Use Wee/Tee to merge stdout/stderr (keep echoText splitted).
```
Comment (aloiscochard) [inline sarsi-elm/Main.hs:25]: I'm currently implementing `sarsi-rs`, and I'll need the same code, I'll try to extract it in the library.

**Comment by aloiscochard** [comment]
It looks great thanks!

Looking forward the demo, next steps before merging:
- [ ] update the `README` accordingly
- [ ] relax upper bounds
- [ ] generalization with `sarsi-hs` and `sarsi-rs`

File: sarsi.cabal [hunk 2]
```diff
@@ -85,6 +85,21 @@ executable sarsi-hs
   ghc-options:          -Wall -dynamic -threaded
   default-language:     Haskell2010
 
+executable sarsi-elm
```
Comment (aloiscochard) [inline sarsi.cabal:88]: Please move this definition before `sarsi-hs` in order to keep things alphabetically ordered (hopefully the list might grow! :) ).

File: sarsi.cabal [hunk 3]
```diff
@@ -85,6 +85,21 @@ executable sarsi-hs
...
+  main-is:              Main.hs
+  build-depends:
+      base
+    , sarsi                 == 0.0.3.0
+    , machines
+    , machines-io
+    , machines-process
+    , process
+    , text
+    , fsnotify              >= 0.2        && < 0.3
```
Comment (aloiscochard) [inline sarsi.cabal:98]: this one shouldn't be needed.

**Comment by aloiscochard** [comment]
@tdroxler I would recommend removing the last commit on a new branch, rebase on master, and then pickup your change and see if you can use https://github.com/aloiscochard/sarsi/commit/fce6d5138ede8fb087f4713f74800588cfb2b5ca to generalize.

Let me know how that sound? ty!

**Comment by tdroxler** [comment]
I can already tell that I can't use `Tools.Shell` because I need `stdout` for elm: https://github.com/aloiscochard/sarsi/pull/7/commits/0d5f2a8cca7060ec13082d5fcd26f6bd173d3f7f#diff-8753902dc76912c00307f477ccf53a08R27

more over I can't simply echo the error message, because I receive a Json and I pretty print it: https://github.com/aloiscochard/sarsi/pull/7/commits/0d5f2a8cca7060ec13082d5fcd26f6bd173d3f7f#diff-8753902dc76912c00307f477ccf53a08R31

**Comment by aloiscochard** [comment]
I see ty, I'll try to add a wrapper that support this case.

**Comment by aloiscochard** [comment]
Superseeded by #11

**Pull request closed without merging**